From 53e0105f87fdaa54873e5c2fcb3a285150459d18 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:36:58 -0400
Subject: [scsi] handle command allocation failure in scsi_reset_provider

Message-id: <1402670238-13916-131-git-send-email-snitzer@redhat.com>
Patchwork-id: 83890
O-Subject: [RHEL7.1 PATCH 130/150] scsi: handle command allocation failure in scsi_reset_provider
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

BZ: 1105204

Upstream commit 95eeb5f5880cd390fd59710f64dc7b84d1e9942f
Author: Christoph Hellwig <hch@lst.de>
Date:   Thu May 1 16:51:05 2014 +0200

    scsi: handle command allocation failure in scsi_reset_provider

    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Reviewed-by: Nicholas Bellinger <nab@linux-iscsi.org>
    Reviewed-by: Mike Christie <michaelc@cs.wisc.edu>
    Reviewed-by: Hannes Reinecke <hare@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/scsi_error.c b/drivers/scsi/scsi_error.c
index d5ab987..799cfe5 100644
--- a/drivers/scsi/scsi_error.c
+++ b/drivers/scsi/scsi_error.c
@@ -2302,6 +2302,12 @@ scsi_reset_provider(struct scsi_device *dev, int flag)
   return FAILED;
 
  scmd = scsi_get_command(dev, GFP_KERNEL);
+ if (!scmd) {
+  rtn = FAILED;
+  put_device(&dev->sdev_gendev);
+  goto out_put_autopm_host;
+ }
+
  blk_rq_init(NULL, &req);
  scmd->request = &req;
 
-- 
1.7.1